autobuild/autobuild_tool_source_environment.py

```diff
@@ -448,25 +448,34 @@ def do_source_environment(args):
         # function, we want to prepend to its existing PATH rather than
         # replacing it with whatever's visible to Python right now.
         try:
-            PATH = vsvars.pop("PATH")
+            vsvars_path = vsvars.pop("PATH")
         except KeyError:
             pass
         else:
             # Translate paths from windows to cygwin format.
             # Match patterns of the form %SomeVar%. Match the SHORTEST such
             # string so that %var1% ... %var2% are two distinct matches.
             percents = re.compile(r"%(.*?)%")
-            PATH = ":".join(
+            existing_path = os.environ["PATH"].split(";")
+            vsvars_path = ":".join(
                 # Some pathnames in the PATH var may be individually quoted --
                 # strip quotes from those.
+                #
                 # Moreover, some may have %SomeVar% substitutions; replace
                 # with ${SomeVar} substitutions for bash. (Use curly braces
                 # because we don't want to have to care what follows.)
                 # may as well de-dup while we're at it
+                #
+                # The vsvars32.bat PATH will contain many redundant paths when
+                # combined with an expanded $PATH when load_vsvars is called.
+                # To avoid truncation problems due to a long PATH each variable
+                # is compared to the live PATH so that we can get the set difference.
+                # This solves the problem of needing to dedupe the path in package
+                # buildscripts.
                 dedup(cygpath("-u", percents.sub(r"${\1}", p.strip('"')))
```

Comment:
why is this dedup function not already doing what we want here?

Reply:
oh, I see now, it is only being applied to the paths that we are adding, even if they are already in "$PATH"

Reply:
Yeah, this is a bit confusing. That function does not do what we want because we are building a shell script that is evaluated later by the build script. The literal bash that is authored is equivalent to `export PATH={path_from_this_method}:$PATH`.